packages/ra-ui-materialui/src/detail/CreateActions.js

```diff
@@ -37,11 +38,14 @@ const sanitizeRestProps = ({
  *         </Create>
  *     );
  */
-const CreateActions = ({ basePath, className, hasList, ...rest }) => (
-    <TopToolbar className={className} {...sanitizeRestProps(rest)}>
-        {hasList && <ListButton basePath={basePath} />}
-    </TopToolbar>
-);
+const CreateActions = ({ className, ...rest }) => {
+    const { basePath, hasList } = useCreateContext();
```

Comment:
you should pass the component props, to provide a fallback in case the context isn't defined (for BC)